packages/nimble-components/src/table/models/table-update-tracker.ts

```diff
@@ -47,70 +33,81 @@ const isColumnInternalsProperty = (
  * Helper class to track what updates are needed to the table based on configuration
  * changes.
  */
-export class UpdateTracker<TData extends TableRecord> {
-    private readonly requiredUpdates: RequiredUpdates = {
-        rowIds: false,
-        groupRows: false,
-        columnIds: false,
-        columnSort: false,
-        columnWidths: false,
-        columnDefinition: false,
-        actionMenuSlots: false,
-        selectionMode: false
-    };
-
-    private readonly table: Table<TData>;
+export class TableUpdateTracker<
+    TData extends TableRecord
+> extends UpdateTracker<
+    [
+        'rowIds',
+        'groupRows',
+        'columnIds',
+        'columnSort',
+        'columnWidths',
+        'columnDefinition',
+        'actionMenuSlots',
+        'selectionMode'
+    ]
+    > {
     private updateQueued = false;
 
-    public constructor(table: Table<TData>) {
-        this.table = table;
+    public constructor(private readonly table: Table<TData>) {
+        super([
+            'rowIds',
+            'groupRows',
+            'columnIds',
+            'columnSort',
+            'columnWidths',
+            'columnDefinition',
+            'actionMenuSlots',
+            'selectionMode'
+        ]);
```

Comment:
Should be able to avoid defining this twice like the TestColumnValidator https://github.com/ni/nimble/blob/14402b0d2e2a8a56454f79aa971324bf96dea5fe/packages/nimble-components/src/table-column/base/tests/table-column.fixtures.ts#L55